### ISSUE TITLE
"kubectl run --restart=Never -ti" doesn't exit interactive session when container exits

### ISSUE THREAD

**Issue by jpetazzo** [kubernetes/kubernetes#105518]
#### What happened:

When running an interactive container with `kubectl run -ti --restart=Never`, when the container exits, `kubectl` doesn't exit. Instead, it keeps running, and needs to be stopped with Ctrl-C.

#### What you expected to happen:

When the process in the container exits, `kubectl run` should exit as well.

#### How to reproduce it (as minimally and precisely as possible):

```bash
kubectl run test$RANDOM -ti --restart=Never --image=alpine
```
When the container comes up, you will see the `#` shell prompt. Type `exit`. `kubectl` should exit, but instead, it keeps running.

#### Anything else we need to know?:

After the container terminates, if we type ENTER, we see the following log line:
```
E1006 17:52:23.093065 1540574 v2.go:105] EOF
```

This appears to be a regression between 1.21 and 1.22:

```
$ kubectl-1.21.5 run test$RANDOM -ti --restart=Never --image=alpine 
If you don't see a command prompt, try pressing enter.
/ # exit
[jp@большой (⎈ |kind-kind:default) bin]
```
^ 1.21.5 behaves correctly.

```
$ kubectl-1.22.0 run test$RANDOM -ti --restart=Never --image=alpine 
If you don't see a command prompt, try pressing enter.
/ # exit

E1006 17:52:23.093065 1540574 v2.go:105] EOF
^C[jp@большой (⎈ |kind-kind:default) bin]
```
^ 1.22.0 requires Ctrl-C to stop.

```
$ kubectl-1.22.1 run test$RANDOM -ti --restart=Never --image=alpine 
If you don't see a command prompt, try pressing enter.
/ # exit

E1006 17:52:38.186082 1540986 v2.go:105] EOF
^C[jp@большой (⎈ |kind-kind:default) bin]
```
^ 1.22.1 requires Ctrl-C too.

#### Environment:

- Kubernetes version (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"archive", BuildDate:"2021-08-19T21:28:12Z", GoVersion:"go1.17", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-21T23:01:33Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration, OS, etc.:

I have reproduced the problem on my ArchLinux machine (with various local clusters like kind, minikube, etc), on Ubuntu 18.04 and 20.04, as well as on Windows 10 with Docker Desktop. In each environment, kubectl 1.21.5 worked fine, but 1.22 exhibited the bug.

**Comment by jpetazzo** [comment]
/sig cli 

♥

**Comment by brianpursley** [comment]
@jpetazzo See my comment here for an explanation and workaround: https://github.com/kubernetes/kubernetes/issues/105554#issuecomment-939577166

TLDR: Will be fixed in kubectl 1.22.3.  Workaround is to set `KUBECTL_COMMAND_HEADERS` environment variable to false
